internal/providers/pluginfw/resources/sharing/resource_share.go

```diff
@@ -0,0 +1,395 @@
+	for _, beforeSdo := range beforeSi.Objects {
+		if _, ok := afterMap[beforeSdo.Name]; ok {
+			continue
+		}
+		changes = append(changes, sharing.SharedDataObjectUpdate{
+			Action:     sharing.SharedDataObjectUpdateActionRemove,
+			DataObject: &beforeSdo,
+		})
+	}
+
+	// not in before so add
+	// if in before but diff then update
+	for _, afterSdo := range afterSi.Objects {
+		if beforeSdo, ok := beforeMap[afterSdo.Name]; ok {
+			if !equal(beforeSdo, afterSdo) {
+				// do not send SharedAs
+				afterSdo.SharedAs = ""
+				changes = append(changes, sharing.SharedDataObjectUpdate{
+					Action:     sharing.SharedDataObjectUpdateActionUpdate,
+					DataObject: &afterSdo,
+				})
+			}
+			continue
+		}
+		changes = append(changes, sharing.SharedDataObjectUpdate{
+			Action:     sharing.SharedDataObjectUpdateActionAdd,
+			DataObject: &afterSdo,
+		})
+	}
+	return changes
+}
+
+func shareChanges(si sharing.ShareInfo, action string) sharing.UpdateShare {
+	var changes []sharing.SharedDataObjectUpdate
+	for _, obj := range si.Objects {
+		changes = append(changes, sharing.SharedDataObjectUpdate{
+			Action:     sharing.SharedDataObjectUpdateAction(action),
+			DataObject: &obj,
+		},
+		)
+	}
+	return sharing.UpdateShare{
+		Name:    si.Name,
+		Owner:   si.Owner,
+		Updates: changes,
+	}
+}
+
+type ShareResource struct {
+	Client *common.DatabricksClient
+}
+
+func (r *ShareResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = pluginfwcommon.GetDatabricksStagingName(resourceName)
+}
+
+func (r *ShareResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attrs, blocks := tfschema.ResourceStructToSchemaMap(ShareInfoExtended{}, func(c tfschema.CustomizableSchema) tfschema.CustomizableSchema {
+		c.SetRequired("name")
+
+		c.AddPlanModifier(stringplanmodifier.RequiresReplace(), "name") // ForceNew
+		c.AddPlanModifier(int64planmodifier.UseStateForUnknown(), "created_at")
+		c.AddPlanModifier(stringplanmodifier.UseStateForUnknown(), "created_by")
+		c.AddPlanModifier(int64planmodifier.UseStateForUnknown(), "object", "added_at")
+		c.AddPlanModifier(stringplanmodifier.UseStateForUnknown(), "object", "added_by")
+
+		c.SetRequired("object", "data_object_type")
+		c.SetRequired("object", "partitions", "values", "op")
+		c.SetRequired("object", "partitions", "values", "name")
+		return c
+	})
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Share",
+		Attributes:  attrs,
+		Blocks:      blocks,
+	}
+}
+
+func (d *ShareResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if d.Client == nil && req.ProviderData != nil {
+		d.Client = pluginfwcommon.ConfigureResource(req, resp)
+	}
+}
+
+func (r *ShareResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	w, diags := r.Client.GetWorkspaceClient()
```

Comment:
Also need to add
`ctx = pluginfwcontext.SetUserAgentInResourceContext(ctx, resourceName)` in each CRUD, ref: https://github.com/databricks/terraform-provider-databricks/blob/main/internal/providers/pluginfw/resources/qualitymonitor/resource_quality_monitor.go#L101

Reply:
👍 e3500130